docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,198 @@
+# thin-edge Domain Model
+
+The following section introduces the **thin-edge** domain model.
+
+**thin-edge** is the glue between the **cloud** and **devices**. I.e. it interfaces on the one hand **devices** (especially resource constrained industrial **devices**), and connects on the other hand the **cloud**.
+The **cloud** could be some IoT Cloud Platform, as e.g. [Cumulocity](https://www.softwareag.cloud/site/product/cumulocity-iot.html), [Azure](https://azure.microsoft.com) or [AWS](https://aws.amazon.com); or an on-premise "IoT system".
+
+**thin-edge** has been designed with the assumption that the **device** is represented in the **IoT cloud** as **device twin**.
+The **device twin** in the IoT cloud is a logical representation of the **device**. As such, it stores the _state_ of the **device**.
+Here _state_ refers to e.g. **Telemetry Data** or **Device Management** details as _installed software versions_ or _configurations_.
+
+## Telemetry Data
+
+**Telemetry data** are _measurements_, _events_ and _alarms_ collected or raised by sensors[^1], actuators[^1] or domain applications[^1].
```

Comment:
I think we are missing a section for the "Inventory data" which is neither measurement nor metrics.

In Cumulocity context this would be classically modelled as a specific fragment (property) on the device managed object. This is partially already mentioned in the first section with the software management example where the installed software packages would be stored somewhere.

Reply:
Applied in new PR, see https://github.com/thin-edge/thin-edge.io/pull/1554#issuecomment-1304053106